spanner/read.go

```diff
@@ -533,11 +539,17 @@ var (
 
 func (d *resumableStreamDecoder) next(mt *builtinMetricsTracer) bool {
 	retryer := onCodes(d.backoff, codes.Unavailable, codes.ResourceExhausted, codes.Internal)
+
+	// Setup and track x-goog-request-id in the manual retries for ExecuteStreamingSql.
+	attempt := uint32(0)
+	riw := d.gsc.generateRequestIDHeaderInjector()
```

Comment:
This should only be initialized once per stream, and not once for each call to `next`. A stream can break halfway, and will be retried from the point where it broke. Now, that will cause a new request number to be generated, instead of registering it as a retry.

See the test in https://github.com/googleapis/google-cloud-go/commit/b5a70e4104fd862ab42eba45ce43f2c3cc7fb5d8

Reply:
Awesome and thank you @olavloite. I added your test and also sent the update so we now get the expected retries and attempt sequences.